### ISSUE TITLE
Context window overflow after ~98 samples due to unbounded gradient accumulation

### ISSUE THREAD

**Issue by Nikhil-Prasad** [zou-group/textgrad#179]
## Description
TextGrad's optimizer accumulates all gradients in memory, causing context window errors when training on datasets with more than ~98 misclassified samples.

## Current Behavior
- Each misclassified sample generates ~600-1000 chars of gradient text
- After 100 samples: ~60k-100k chars → ~15k-25k tokens
- Optimizer prompt includes ALL historical gradients
- Crashes with "context_length_exceeded" errors

## Expected Behavior
System should handle arbitrary dataset sizes without context overflow.

## Reproduction
```python
# Train on 200+ samples with ~50% error rate
optimizer = TextualGradientDescent(parameters=[prompt])
for i in range(200):
    loss.backward()  # Generates gradient
    optimizer.step()  # Crashes around i=98
```

## Related to Gradient-RAG Implementation
This is the core issue that Gradient-RAG aims to solve by implementing semantic retrieval of gradients instead of unbounded accumulation.
